Copilot/find instances of cite (#294)
symbiotic-engineering/MDOcean: pubs/applied-ocean-research-model/sections/module-details.tex

\ifdefined\DISSERTATION
    Methods for WEC dynamics and control simulation in an optimization context were reviewed in \Cref{sec:lit} and include time-stepping (nonlinear, numerical, high computation cost), frequency domain (linear, analytical, low computation cost), pseudo-spectral (nonlinear, numerical, medium computation cost), quasi-linearized pseudo-spectral techniques (approximate nonlinear, semi-analytical, low computational cost), and Pontryagin's maximum principle (nonlinear, analytical, low computational cost).

>     In wave energy, both spectral-domain numerical methods and analytical methods have generated recent research interest, with studies \citet{bacelli_numerical_2015,coe_initial_2020,tan_computationally_2026,nie_optimal_2016,da_silva_stochastic_2023} and studies \citet{abdulkadir_optimal_2024,lin_fast_2025} serving as illustrative examples of each case.

Author: ```suggestion
    In wave energy, both spectral-domain numerical methods and analytical methods have generated recent research interest. \citet{bacelli_numerical_2015,coe_initial_2020,tan_computationally_2026,nie_optimal_2016,da_silva_stochastic_2023} study the former and \citet{abdulkadir_optimal_2024,lin_fast_2025} illustrate the latter.
```

This synthesis can occur in a number of ways.
If a time-domain optimal control law (e.g. those that \citet{abdulkadir_optimal_2024,lin_fast_2025} develop via the Pontryagin Maximum Principle (PMP)) is available for the constraint under consideration (M1.1), it can be used as the nonlinear optimal controller directly, essentially bypassing the use of QLPS in the control synthesis step (\Cref{fig:mod-freq-domain-synthesis}) and using it only for evaluation (\Cref{fig:mod-freq-domain-evaluation}).
\ifdefined\DISSERTATION

>     This differs from the standard PMP method in that the evaluation of the response and power corresponding to the time-domain PMP controller is determined in the frequency domain, rather than via numerical time-integration (as in the method of \citep{zou_optimal_2017}) or analytical time-integration (as in the method of \citep{lin_fast_2025}).

Author: ```suggestion
    This differs from the standard PMP method in that the evaluation of the response and power corresponding to the time-domain PMP controller is determined in the frequency domain, rather than via numerical time-integration (as in the method of \citet{zou_optimal_2017}) or analytical time-integration (as in the method of \citet{lin_fast_2025}).
```

\ifdefined\DISSERTATION
    The constrained optimization to identify the controller in (M1.2) and (M2) can be performed numerically or analytically by analyzing an impedance-mismatched linear system in the frequency domain and will be discussed in detail in \Cref{sec:optimal-control}.
    The current MDOcean software uses the numerical approach, which requires more computation time but is simpler to implement and equally accurate.

>     The analytical approach is nonetheless derived in \Cref{sec:optimal-control} (with further details and graphical visualization of the constraints in \Cref{sec:appendix-qp-solution} and reference \citep{mccabe_force-limited_2024}) and recommended for future implementation.

Author: ```suggestion
    \Cref{sec:optimal-control} derives the analytical approach, while \citet{mccabe_force-limited_2024} and \Cref{sec:appendix-qp-solution} provide details and visualizations.
```

This approach neglects the corresponding effect of the saturation/zeroing on other quantities in the system, which lowers the accuracy, and the controller does not have the opportunity to even approximately consider the constraint, resulting in a power production potentially much lower than the constrained optimal.
\ifdefined\DISSERTATION
    This simple approach does not require any derivations to implement.

>     It has been used in several previous WEC optimization studies for various constraints \citet{garcia-teruel_design_2022,garcia-teruel_reliability-based_2021,cotten_multi-objective_2022,mccabe_constrained_2013} and is evidently the most common quasi-linearized pseudo-spectral method in the marine energy field.

Author: ```suggestion
    It has been used in several previous WEC optimization studies for various constraints \citep{garcia-teruel_design_2022,garcia-teruel_reliability-based_2021,cotten_multi-objective_2022,mccabe_constrained_2013} and is evidently the most common quasi-linearized pseudo-spectral method in the marine energy field.
```

\else
    This simple approach has been used in several previous WEC optimization studies for various constraints in the studies \citep{garcia-teruel_design_2022,garcia-teruel_reliability-based_2021,cotten_multi-objective_2022,mccabe_constrained_2013}
    and is the most common quasi-linearized pseudo-spectral method in the marine energy field.
\fi
It is utilized here for the power limit.

As a last resort, the energy in that sea state can be zeroed (M4),

> essentially assuming that the device enters survival mode in those cases to avoid constraint violation, or the entire design can be marked infeasible with a constraint in the optimization (M5), which reference~\citep{mccabe_constrained_2013} pursues for a radiation limit constraint.

Author: ```suggestion
essentially assuming that the device enters survival mode in those cases to avoid constraint violation, or the entire design can be marked infeasible with a constraint in the optimization (M5), which \citet{mccabe_constrained_2013} pursues for a radiation limit constraint.
```

    Because the same describing function is applied in both synthesis and evaluation, the inverse and forward describing functions cancel; the time-domain nonlinear controller need not be reconstructed for power evaluation, only for hardware implementation.
\fi
While the describing function method used here has been known for over half a century, MDOcean represents its first application in a wave energy optimization or open-source simulation, to the authors' knowledge.

> The RAFT open-source Python package for offshore wind turbines uses the drag describing function with iteration but does not include describing functions for saturation \citep{hall_open-source_2022}.

Author: ```suggestion
\citet{hall_open-source_2022} documents the RAFT open-source Python package for offshore wind turbines, which uses the drag describing function with iteration but does not include describing functions for saturation.
```

impedance ($\mathbf{Z}$) groups all ports together;
cascade ($\mathbf{a}$/ABCD) transmits efforts and flows sequentially between port groups and is used for the PTO dynamics (\Cref{sec:appendix-pto-model});
hybrid ($\mathbf{h}$) mixes the two and is used for the PTO kinematics.

> Full conventions, conversion formulas, and the cascade inversion rule are given in \Cref{sec:appendix-multiport}, following \citep{reveyrand_multiport_2018}.

Author: ```suggestion
Full conventions, conversion formulas, and the cascade inversion rule are given in \Cref{sec:appendix-multiport}, following \citet{reveyrand_multiport_2018}.
```

    Note that this is not strictly equivalent to assuming the generator has constant efficiency, since controllers with reactive terms ($K_l\neq 0$) spend some time expending energy, which reverses the direction of the generator efficiency.
    The average efficiency is assumed to also apply to the peak powers, though this is not guaranteed in general for the same reason.
\fi

> Validation in \Cref{sec:validation-benchmarking} sets $\eta=80\%$ to match the RM3 design assumption of \citep{RM3}.

Author: ```suggestion
Validation in \Cref{sec:validation-benchmarking} sets $\eta=80\%$ to match the RM3 design assumption of \citet{RM3}.
```

    higher-order nonlinear potential flow effects, and
    steady and unsteady viscous separation phenomena are all neglected.
    In this section, we model the form drag due to viscous flow separation, including steady wake losses, turbulence, and cycle-averaged vortex shedding.

>     These effects have been found to be the dominant drag force for heaving WECs \citet{quartier_influence_2021} and are important to avoid unrealistically high resonant peaks in the response amplitude which would lead to overestimates of power production.

Author: ```suggestion
    \citet{quartier_influence_2021} finds that these effects are the dominant drag force for heaving WECs and are important to avoid unrealistically high resonant peaks in the response amplitude which would lead to overestimates of power production.
```

% Maha question: should we just jump to the result instead of deriving p_d approximation? (5/18/26)
Drag force $\vec{F}_d$ is proportional to the product of the relative velocity of the WEC and incident wave, $\vec{v}_{\text{rel}}(y,t)=\vec{\dot{\xi}}(t)-\vec{v}_{\text{wave}}(y,t)$, with its absolute value.
The resulting quadratic form $|\vec{F}_d|\sim|\vec{v}_{\text{rel}}|^2$ implies a nonlinear input-output response that would normally require time-domain simulation.

> To improve computational efficiency and allow solution in the frequency domain, the nonlinearity is quasi-linearized with a describing function, following \citep{quartier_influence_2021}.

Author: ```suggestion
To improve computational efficiency and allow solution in the frequency domain, the nonlinearity is quasi-linearized with a describing function, following \citet{quartier_influence_2021}.
```

To find the long-term average power production $\overline{P}_{\text{elec}}$ in a location with a given distribution of sea states, the power matrix $\mathbf{P}^{H,T}_{\text{elec}}$ for each sea state is weighted by that sea state's probability using a Joint Probability Density (JPD) matrix and then summed.
This method is illustrated in \Cref{fig:JPD-multiply}.

> The JPD used in this analysis represents wave conditions in Humboldt Bay, CA and is taken from the study by \citep{janzou_system_2020}.

Author: ```suggestion
The JPD used in this analysis represents wave conditions in Humboldt Bay, CA and is taken from the study by \citet{janzou_system_2020}.
```

\ifdefined\DISSERTATION
    The MDOcean economics model aims not to predict the precise cost breakdown of a given design but to accurately reflect the scaling of cost with design variables to facilitate design optimization and tradeoff analysis.
    With this in mind, the model lumps costs into categories that are either constant or scale with design, and the constant costs and scale factors are tuned to match more detailed cost breakdowns.

>     The cost data chosen for tuning is the RM3 Cost Breakdown Structure (CBS) \citep{neary_reference_2014} (\Cref{tab:CBS} in \Cref{sec:appendix-econ}).

Author: ```suggestion
    The cost data chosen for tuning is the RM3 Cost Breakdown Structure (CBS), first published by \citet{neary_reference_2014} and reproduced in \Cref{tab:CBS} of \Cref{sec:appendix-econ}.
```
